dspace-api/src/main/java/org/dspace/content/dao/impl/ItemDAOImpl.java

```diff
@@ -139,27 +142,77 @@ protected ItemDAOImpl() {
         return iterate(query);
     }
 
-    enum OP { equals, not_equals, like, not_like, contains, doesnt_contain, exists, doesnt_exist, matches,
-        doesnt_match }
```

Comment:
The DSpace Rest Report tools rely extensively on the matches and doesnt_match operators.  Without those operators, most of the benefits of those tools are lost.  See https://terrywbrady.github.io/restReportTutorial/intro

Reply:
To clarify (as I misunderstood myself in Slack #dev discussion)... @terrywbrady is asking if we can:
1. Revert all changes to `OP` in this PR
2. Revert all changes to the `findByMetadataQuery()` method in this PR
3. Change the `findByMetadataQuery()` method to Deprecated (as it would still be dependent on deprecated Hibernate functionality). 

The reason for this is that `findByMetadataQuery()` is ONLY used by the existing, deprecated REST API (v6).  So, deprecating this method allows the deprecated REST API to work "as-is".  If we cannot find a replacement for this functionality in DSpace 7, then we simply drop the `findByMetadataQuery()` method -- or we implement a partial replacement (which may not include `matches` or `doesnt_match`)

Reply:
This looks good.

We should also 
- revert the change to rest.cfg